src/scheduler/ExecutableIntervalSchedule.ts

```diff
@@ -0,0 +1,65 @@
+import humanInterval from 'human-interval';
+import { max } from 'lodash';
+import { DateTime } from 'luxon';
+import { IntervalSchedule } from '../job/MomoJob';
+import { momoError } from '../logging/error/MomoError';
+import { TimeoutHandle, setSafeIntervalWithDelay } from '../timeout/setSafeIntervalWithDelay';
+import { Logger } from '../logging/Logger';
+import { ExecutableSchedule, NextExecutionTime } from './ExecutableSchedule';
+import { ExecutionInfo } from '../job/ExecutionInfo';
+
+export class ExecutableIntervalSchedule implements ExecutableSchedule<IntervalSchedule> {
+  private readonly interval: string;
+  private readonly firstRunAfter: number;
+  private timeoutHandle?: TimeoutHandle;
+
+  constructor({ interval, firstRunAfter }: IntervalSchedule) {
+    this.interval = interval;
+    this.firstRunAfter = firstRunAfter;
+  }
+
+  toObject(): IntervalSchedule {
+    return { interval: this.interval, firstRunAfter: this.firstRunAfter };
+  }
+
+  execute(callback: () => Promise<void>, logger: Logger, executionInfo?: ExecutionInfo): NextExecutionTime {
+    const interval = this.parse();
+    const delay = this.calculateDelay(interval, executionInfo);
+
+    this.timeoutHandle = setSafeIntervalWithDelay(callback, interval, delay, logger, 'Concurrent execution failed');
+
+    return { date: DateTime.fromMillis(DateTime.now().toMillis() + delay) };
```

Comment:
is there a reason to wrap the date into an object? I guess the method could simply use `Date` as a return type?
Or do you argue that giving the interface a speaking name (`NextExecutionTime`) improves readability and that wins over the using simple types?

Reply:
However, in that case the property `date` should maybe be renamed to "nextExecution", then line 80 in JobScheduler also becomes easy to read

Reply:
Yeah, exactly. I wanted a better name for it - especially if we expand it in the future. But the rename is a good idea.